sqlest/src/main/scala/sqlest/ast/Column.scala

```diff
@@ -119,6 +119,41 @@ sealed trait AliasedColumn[A] extends Column[A] with CellExtractor[ResultSet, A]
 
 }
 
+/**
+ * A column that has an index associated with it.
+ *
+ * This column is only for internal use when accessing generated keys
+ *
+ */
+case class IndexedColumn[A](index: Int)(implicit val columnType: ColumnType[A]) extends Column[A] with CellExtractor[ResultSet, A] {
```

Comment:
I don't think that @p14n wants the developer to have to have to supply the column alias for which the key is returned, hence the new column that reads by index?